### PR TITLE
Update documentation on parsing idioms

### PR DESCRIPTION
**Pull request by ltratt**

### CODE REVIEW

File: doc/src/parsing_idioms.md [hunk 1]
```diff
@@ -108,6 +123,69 @@ fn map_err(r: Result<Lexeme<StorageT>, Lexeme<StorageT>>)
 ```
 
 
+### Using `Err(Box<dyn Error>)`
+
+The idea here is that if we both stop evaluating normal action code, and
```
Comment (ptersilie) [inline doc/src/parsing_idioms.md:128]: remove the `if`

Comment (ltratt) [reply doc/src/parsing_idioms.md]: Fixed in https://github.com/softdevteam/grmtools/pull/198/commits/211b301994666bcb5e2b7751efb7aa10b820afde.

**Comment by ptersilie** [comment]
Please squash.

**Comment by ltratt** [comment]
Squashed.

**Pull request merged**